src/Extension/CoreExtension.php

```diff
@@ -1374,6 +1374,25 @@ function twig_array_batch($items, $size, $fill = null, $preserveKeys = true)
  */
 function twig_get_attribute(Environment $env, Source $source, $object, $item, array $arguments = [], $type = /* Template::ANY_CALL */ 'any', $isDefinedTest = false, $ignoreStrictCheck = false, $sandboxed = false, int $lineno = -1)
 {
+    // deep attribute
+    if(\is_string($item)) {
+        list($item, $deepItem) = \array_merge(\explode("->", $item, 2), [null]);
```

Comment:
Detecting the deep case based on a `->` delimiter in the string passed to `twig_get_attribute` is a BC break, because it forbids using that in the name of an attribute (and array keys allow *any* string so you won't be able to find a safe magic delimiter)